### PR TITLE
feat: add batch operations and distributed lock support

### PR DESCRIPTION
**Pull request by able-wong**
## Summary

- Add `setIfNotExists(key, value, ttl?)` - SETNX for distributed locks/mutex patterns
- Add `getMany<T>(keys)` - MGET for batch fetching in single round trip
- Add `setMany<T>(entries, ttl?)` - MSET for batch setting with optional TTL
- Add `expire(key, ttl)` - Update TTL without changing value
- Enhance `getOrSet` with `isValid` validator for staleness checks
- Extract `serialize()` helper for DRY
- Bump version to 0.3.0

## New API Methods

| Method | Redis Command | Use Case |
|--------|--------------|----------|
| `setIfNotExists(key, value, ttl?)` | `SET ... NX` | Distributed locks, deduplication |
| `getMany<T>(keys)` | `MGET` | Batch fetching (single round trip) |
| `setMany<T>(entries, ttl?)` | `MSET` + pipeline `EXPIRE` | Batch setting |
| `expire(key, ttl)` | `EXPIRE` | Update TTL without changing value |

## Important Notes

**`setIfNotExists` and mutex patterns:**
In graceful mode (default), returns `false` both when key exists AND when cache is unavailable. For mutex/lock patterns where this distinction matters, use `{ onError: 'throw' }`.

## Test plan

- [x] Unit tests for all new methods (18 new tests)
- [x] Tests for throw mode
- [x] Tests for object/complex value serialization
- [x] Tests for edge cases (empty arrays, expired keys)
- [x] Lint passes
- [x] Build passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -248,6 +303,16 @@ const client = new ResilientCacheClient({
 | `getOrSet<T>(key, factory, ttlSeconds?, options?)` | factory result | throws `CacheUnavailableError` |
 | `exists(key, options?)` | `false` | throws `CacheUnavailableError` |
 | `ttl(key, options?)` | `-2` | throws `CacheUnavailableError` |
+| `setIfNotExists<T>(key, value, ttlSeconds?, options?)` | `false` | throws `CacheUnavailableError` |
+| `getMany<T>(keys, options?)` | `(null)[]` | throws `CacheUnavailableError` |
```
[the remark this answers is not in the document]
Comment (able-wong) [reply README.md]: Fixed in 4c1ab98 - changed to `null[] (same length as input)`

File: src/MockCacheClient.ts [hunk 2]
```diff
@@ -322,6 +341,120 @@ export class MockCacheClient implements ICacheClient {
...
+    }
+
+    if (this.isExpired(key)) {
+      return false;
+    }
+
+    const item = this.store.get(key);
+    if (!item) {
+      return false;
+    }
```
[the remark this answers is not in the document]
Comment (able-wong) [reply src/MockCacheClient.ts]: Fixed in 4c1ab98 - removed redundant check, added non-null assertion with explanatory comment

File: src/ResilientCacheClient.ts [hunk 3]
```diff
@@ -454,6 +469,136 @@ export class ResilientCacheClient implements ICacheClient {
...
+    }
+    if (ttlSeconds !== undefined) {
+      this.validateTtl(ttlSeconds);
+    }
+    return this.executeCommand('setMany', false, options, async () => {
+      // Build key-value pairs for MSET
+      const args: string[] = [];
+      for (const entry of entries) {
+        args.push(entry.key, this.serialize(entry.value));
+      }
```
[the remark this answers is not in the document]
Comment (able-wong) [reply src/ResilientCacheClient.ts]: Fixed in 4c1ab98 - replaced for loop with `flatMap`
